crates/turborepo-lib/src/commands/link.rs

```diff
@@ -127,56 +126,125 @@ pub async fn link(base: &mut CommandBase, modify_gitignore: bool) -> Result<()>
-    };
+            let team_id = match selected_team {
+                SelectedTeam::User => user_response.user.id.as_str(),
+                SelectedTeam::Team(team) => team.id.as_str(),
+            };
 
-    if modify_gitignore {
-        add_turbo_to_gitignore(base)?;
-    }
+            verify_caching_enabled(&api_client, team_id, token, Some(selected_team.clone()))
+                .await?;
 
-    println!(
-        "
-{}  Turborepo CLI authorized for {}
+            fs::create_dir_all(base.repo_root.join(".turbo"))
+                .context("could not create .turbo directory")?;
+            base.repo_config_mut()?
+                .set_team_id(Some(team_id.to_string()))?;
 
-{}
-    ",
-        base.ui.rainbow(">>> Success!"),
-        base.ui.apply(BOLD.apply_to(chosen_team_name)),
-        GREY.apply_to("To disable Remote Caching, run `npx turbo unlink`")
-    );
-    Ok(())
+            let chosen_team_name = match selected_team {
+                SelectedTeam::User => user_display_name,
+                SelectedTeam::Team(team) => team.name.as_str(),
+            };
+
+            if modify_gitignore {
+                add_turbo_to_gitignore(base)?;
+            }
+
+            println!(
+                "
+    {}  Turborepo CLI authorized for {}
+
+    {}
+        ",
+                base.ui.rainbow(">>> Success!"),
+                base.ui.apply(BOLD.apply_to(chosen_team_name)),
+                GREY.apply_to("To disable Remote Caching, run `npx turbo unlink`")
+            );
+            return Ok(());
+        }
+        LinkTarget::Spaces => {
+            println!(
+                ">>> Vercel Spaces (Beta)
+
+      For more info, see {}
+      ",
+                base.ui.apply(UNDERLINE.apply_to(SPACES_URL))
+            );
+
+            if !should_link_spaces(base, &repo_root_with_tilde)? {
+                return Err(anyhow!("canceled"));
+            }
+
+            let spaces_response = api_client
+                .get_spaces(token, base.repo_config()?.team_id())
+                .await
+                .context("could not get spaces information")?;
+
+            let selected_space = select_space(base, &spaces_response.spaces)?;
+
+            // print result from selected_space
+            let space = match selected_space {
+                SelectedSpace::Space(space) => space,
+            };
+
+            let persisted_space_id = add_space_id_to_turbo_json(base, &space.id);
+
+            match persisted_space_id {
+                Ok(_) => {
+                    println!(
+                        "
+    {} {} linked to {}
+
+    {}
+        ",
+                        base.ui.rainbow(">>> Success!"),
+                        base.ui.apply(BOLD.apply_to(&repo_root_with_tilde)),
+                        base.ui.apply(BOLD.apply_to(&space.name)),
+                        GREY.apply_to(
+                            "To remove Spaces integration, run `npx turbo unlink --target spaces`"
```

Comment:
if we know the user invoked global turbo, we should change this to a global turbo command `turbo unlink`. (not sure if we have any affordances for this, and not blocking)